paddle/fluid/framework/prune.cc

```diff
@@ -189,6 +194,20 @@ void prune_impl(const proto::ProgramDesc& input, proto::ProgramDesc* output,
             }
           }
         }
+
+        // Recurrent op's states are also dependent vars
+        if (op->type() == kRecurrent) {
```

Comment:
A bit hard code. Maybe we should find a better way later after this PR merges.

Reply:
I don't think it is hard code here ... because due to early design, States are Attrs of Recurrent OP. There is no way I can avoid adding States as dependent vars. This is the only way I can fix the bug on current design framework.

As we have discussed, changing Attrs to Inputs/Outputs is not encouraged in PaddlePaddle and it may bring new bugs. We can rewrite RNNs in the future and discard early design, but now it's too risky for release 1.6 ...

In addition, I'm also suffering from the early design and TU CAO it.